Documentation/Using failure models.md

```diff
@@ -0,0 +1,83 @@
+# Using failure model example
+
+In some cases your request might return a server error, but you may still want to parse the data into a failure model.
+
+For example, the server returns response code 400, with JSON body:
+
+```json
+{
+  "errorCode": "incorrectPassword"
+}
+```
+
+TAB resource loader will still return a `.failure` case for this request but it may still contain the required model.
```

Comment:
this should probably be more specific, saying that the network service will return a `NetworkResponse` enum with a `failure` case